### ISSUE TITLE
Windows | HTTP Server for metrics experiences high latency and high CPU

### ISSUE THREAD

**Issue by pcjonathan** [fluent/fluent-bit#4711]
## Bug Report

**Describe the bug**
When using the HTTP server for metrics and health check, fluent-bit will be unable to find available workers to respond to the request, causing requests to take several seconds and significant processor usage. It will reply instantly to begin with however after a short period, it'll start being highly latent. After 24 hours on a box with zero activity and zero logs, we observed API request times of 15+ seconds and from negligible to 30% CPU usage, which was only increasing over **time.**

**To Reproduce**
- Be on Windows
- Enable http server: `http_server  On`
- Make requests to http server; any should do, be it metrics, healthcheck, prometheus, etc, e.g. just http://localhost:2020/api/v1/metrics in the browser will do. This will begin the log spam.
- Continue performing requests and after a while (we found ~30-60 mins at ~2 per minute), impact will start occurring and grow worse over time.

**Expected behavior**
Should reply instantly without high cpu usage

**Screenshots**
No screenshots but the console will output tens of lines per second of `[Warning] [server] Over capacity.`, which is here:

https://github.com/fluent/fluent-bit/blob/ab2a47fcb64ec111e0abf826388a25d444cf92f7/lib/monkey/mk_server/mk_server.c#L414 

**Your Environment**

* Version used: Tested on 1.8.2, 1.8.11, 1.8.12
* Operating System and version: Windows (tested on 10, 11, Server 2012 R2)
* Appears Windows-only
* Filters and plugins: HTTP Server

**Docker Image Test**
I tested the Docker Image via WSL2 using our original config, an edited version of the default provided in the documentation (as the input in the default in incompatible with Windows), and a stripped down one for below.

Windows CLI (Broken):
`PS D:\td-agent-bit-1.8.12-win64> .\bin\fluent-bit.exe -c "conf\fluent-bit.conf"`

WSL2 Docker Run (Working):
`docker run -p 127.0.0.1:20200:20200 -v "D:\td-agent-bit-1.8.12-win64\conf\fluent-bit.conf:/fluent-bit/etc/fluent-bit.conf" fluent/fluent-bit:1.8`

Here is the short config file used:

```
[SERVICE]
    flush        5
    daemon       Off
    log_level    info
    http_server  On
    http_listen  0.0.0.0
    http_port    20200

[INPUT]
    name random
    tag  random.local

[OUTPUT]
    name  stdout
    match *
```

**Additional context**
We're wanting to implement fluent-bit in our environment of Windows 2012 R2 servers so we can forward logs from our application to Datadog to replace our existing logging. We wish to have use the api to perform health checks via PRTG and further metrics via Prometheus/Grafana. I'm not sure what affect this would have on the underlying logs being forwarded because high box impact and possible timeouts on monitoring is a blocking problem.
Many thanks.

**Comment by patrick-stephens** [comment]
This was the 64-bit version as well.

**Comment by leonardo-albertovich** [comment]
This issue should be addressed by PRs #4886 and #4887.

**Comment by patrick-stephens** [comment]
@pcjonathan did those help?
